JeongJaeHeon/app.js

```diff
@@ -0,0 +1,161 @@
+const http = require('http');
+
+
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+
+const { DataSource } = require('typeorm')
+
+const app = express();
+
+const mysqlDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE,
+})
+
+mysqlDataSource.initialize()
+    .then(() => {
+        
+        console.log("Data Source has been initialized")
+    
+    })
+    .catch(() => {
+        
+        console.log("Failed to initialize Data source")
+    
+    })
+
+app.use(express.json());
+app.use(cors());
+app.use(morgan('dev'));
+
+app.get('/ping',(req, res, next) => {
+    res.status(200).json({message : 'pong'});
+});
+
+app.get('/posts/:userId', async(req, res) => {
+    const { userId } = req.params;
+
+    const posts = await mysqlDataSource.query(
+        `SELECT 
+            u.id AS userId,
+            u.profile_image AS userProfileImage,
+            p.id AS postingId,
+            p.image_url AS postingImageUrl,
+            p.content AS postingContent
+          FROM posts p
+          JOIN users u ON p.user_id = u.id
+          WHERE p.user_id = ?
+        `, [ userId ])
+        res.status(200).json(posts);
+    })
+
+app.get('/posts', async(req, res) => {
+
+    const posts = await mysqlDataSource.query(
+        `SELECT
+          u.id AS userId,
+          u.name AS userName,
+          p.id AS postingId,
+          p.title AS postingTitle,
+          p.content AS postingContent
+        FROM posts p
+        JOIN uses u ON p.user_id = u.id;
+        `) 
+        res.status(200).json(posts)
+    })
+    
+app.post('/users/likes', async(req, res) => {
+    const { userId, postId } = req.body
+
+    await mysqlDataSource.query(
+        `UPDATE likes
+          SET 
+            user_id = ?, 
+            post_id = ? 
+          `, [ userId, postId ])
+    res.status(200).json({message : "likes Created"})
+})
```

Comment:
엔드포인트는 특정 기능이 담길 수 있도록 작성해주면 됩니다!

지금 작성해주신 부분도 좋지만 좋아요는 사실 로그인을 할 수 있는 기능이고, 추후 이 부분에 대해서 로그인을 확인하는 모듈을 작성해볼 수 있습니다.

그렇기에 다음과 같이 수정할 수 있습니다.

```suggestion
app.post('/likes', async(req, res) => {
    const { userId, postId } = req.body

    await mysqlDataSource.query(
        `UPDATE likes
          SET 
            user_id = ?, 
            post_id = ? 
          `, [ userId, postId ])
    res.status(200).json({message : "likes Created"})
})
```